addons/velero/1.6.2/install.sh

```diff
@@ -38,64 +67,100 @@ function velero_join() {
 function velero_install() {
     local src="$1"
     local dst="$2"
+    local has_secret
 
+    # TODO: remove this from the migration
     # Pre-apply CRDs since kustomize reorders resources. Grep to strip out sailboat emoji.
-    $src/assets/velero-v${VELERO_VERSION}-linux-amd64/velero install --crds-only | grep -v 'Velero is installed'
+    "$src"/assets/velero-v"${VELERO_VERSION}"-linux-amd64/velero install --crds-only | grep -v 'Velero is installed'
 
     local resticArg="--use-restic"
     if [ "$VELERO_DISABLE_RESTIC" = "1" ]; then
         resticArg=""
     fi
 
+    # detect if we need to use object store or pvc
     local bslArgs="--no-default-backup-location"
     if ! kubernetes_resource_exists "$VELERO_NAMESPACE" backupstoragelocation default; then
-        bslArgs="--provider aws --bucket $VELERO_LOCAL_BUCKET --backup-location-config region=us-east-1,s3Url=${OBJECT_STORE_CLUSTER_HOST},publicUrl=http://${OBJECT_STORE_CLUSTER_IP},s3ForcePathStyle=true"
+        bslArgs="--provider replicated.com/pvc --bucket velero-internal-snapshots --backup-location-config storageSize=${VELERO_PVC_SIZE},resticRepoPrefix=/var/velero-local-volume-provider/velero-internal-snapshots/restic"
```

Comment:
this BSL can be changed only when disableS3 flag is specified in the kurl spec?

Reply:
No, I think this condition basically targets only new installs. If there is any existing BSL, like one created through KOTS, then we pass `--no-default-backup-location` and velero won't touch the existing one. IICR from discussions with @ajp-io during planning, we want to move over all new installs of kurl to the new internal storage mechanism immediately.

Reply:
We've changed some of our decisions since then, so now I wonder what is best. If we make all new installs use PVCs for everything, we will end up with the split architecture that Marc mentioned where customers could have end-customers deployed differently depending on whether they initially deployed before or after this change. He wasn't a fan of that. Makes me wonder if we should just respect the `disableS3` flag (or whichever new flag we might use) instead, even for new installs. WDYT, Dan?

Reply:
Yeah, I guess it makes sense not to change the existing behavior unless you add a flag. Do we want to gate this with a feature flag for a while too while we backport to other velero versions and maybe figure out if there is enough storage in the cluster? I can change the environment variable `BETA_VELERO_MIGRATE_FROM_OBJECT_STORE` to something more generic like `BETA_VELERO_USE_INTERNAL_PVC `